Coherence Explanation (#38)
MrLongbottom/P10-Article: sections/Experiment.tex

	\item Word set confirmation measure
	\item Aggregation of confirmation measures
\end{enumerate}

> 

Reviewer: ```suggestion
\vejleder[inline]{hvad er intuitionen? hvorfår gør man det på denne måde?}
```

> The equations, explanations and values of the hyperparameters in this section are based on~\citet{Syed2017coherence} and the \textit{gensim} python package\footnote{\url{https://radimrehurek.com/gensim/}}.

Reviewer 2: ```suggestion
The equations, explanations, and values of the hyperparameters in this section are based on~\citet{Syed2017coherence} and the \textit{gensim} python package\footnote{\url{https://radimrehurek.com/gensim/}}.
```

> For segmentation a set of word pairs $S$ is created, which pairs each word in the top-N most probable words $W$ in a specific topic $t$ with all the other words in $W$.

Reviewer 2: ```suggestion
For segmentation, a set of word pairs $S$ is created, which pairs each word in the top-N most probable words $W$ in a specific topic $t$ with all other words in $W$.
```

$S$ is defined by \autoref{eq:set_of_word_pairs}.

\begin{equation}\label{eq:set_of_word_pairs}
	S = \{(W', W*)|W' = \{w_i\};w_i \in W;W* = W\}
\end{equation}

> Before calculating word probabilities, a sliding window of size $s = 110$ is used to create a set of sub documents $D_s$ over the document set $D$.

Reviewer: ```suggestion
Before calculating word probabilities, a sliding window of size $s$ where $s =110$ is used to create a set of sub documents $D_s$ over the document set $D$.
```

These sub documents are used rather than the normal documents to capture some degree of words proximity.

> Word probabilities are calcuated based on how many documents within the document set $D_s$ that they occour in.

Reviewer: ```suggestion
Word probabilities are calculated based on how many documents within the document set $D_s$ that they occur in.
```

$p(w_i)$, is the number of sub documents in which the word $w_i$ occurs divided by $|D_s|$.
$p(w_i, w_j)$, is the number of sub documents in which both words occur divided by $|D_s|$. 

> We create a \gls{npmi} matrix of size $|W|\times|W|$, with one entry word pair combination in $W$.

Reviewer: ```suggestion
We create a \gls{npmi} matrix of size $|W|\times|W|$, with every entry word pair combination in $W$.
```

Author: ```suggestion
We create a \gls{npmi} matrix of size $|W|\times|W|$, with one entry per word pair combination in $W$.
```

\begin{equation}\label{eq:coherence_2}
	\text{NPMI}(w_i,w_j) =  \frac{\log\frac{p(w_i,w_j) + \epsilon}{p(w_i)\cdot p(w_j)}}{-log(p(w_i,w_j) + \epsilon)}
\end{equation}
\noindent where $\epsilon$ is a low number ($10^{-12}$) used to avoid $log(0)$.
The \gls{npmi} matrix describes how much each word occurs alone versus together with other words.
%This is a -1 to 1 normalization.

> After having calculated the \gls{npmi} matrix, we construct context vectors for both element $W'$ and $W*$ in each word-pair $S_i$, by summing over the rows of the \gls{npmi} matrix.

Reviewer: ```suggestion
After having calculated the \gls{npmi} matrix, we construct context vectors for both elements $W'$ and $W*$ in each word-pair $S_i$, by summing over the rows of the \gls{npmi} matrix.
```

This summation describes how much each top word co-occurs with the other words in $W$.
\begin{equation}\label{eq:coherence_1}
	\overrightarrow{v}(W') = \left\{ \sum_{w_i \in W'} \text{NPMI}(w_i, w_j)^{\gamma} \right\}_{j=1,\dots,|W|}
\end{equation}

We now have a pair of context vectors for each word pair $S_i$ and we want to know how different these vectors are.
This is calculated using cosine similarity as a confirmation measure.
\begin{equation}\label{eq:coherence_3}
	\phi_{S_i}(\overrightarrow{u}, \overrightarrow{w}) = \frac
	{\sum_{i = 1}^{|W|} u_i \cdot w_i}
	{\|\overrightarrow{u}\|_2 \cdot \|\overrightarrow{w}\|_2}
\end{equation}

\noindent where $\overrightarrow{u}$ is the context vector $\overrightarrow{v}(W')$, and $\overrightarrow{w}$ is the context vector $\overrightarrow{v}(W^*)$.

> Lastly the confirmation measures are aggregated using the arithmic mean, to achieve the coherence value of topic $t$

Reviewer: ```suggestion
Lastly, the confirmation measures are aggregated using the arithmetic mean, to achieve the coherence value of topic $t$
```

	\item Aggregation of confirmation measures
\end{enumerate}
\vejleder[inline]{hvad er intuitionen? hvorfår gør man det på denne måde?}
The equations, explanations and values of the hyperparameters in this section are based on~\citet{Syed2017coherence} and the \textit{gensim} python package\footnote{\url{https://radimrehurek.com/gensim/}}.

For segmentation a set of word pairs $S$ is created, which pairs each word in the top-N most probable words $W$ in a specific topic $t$ with all the other words in $W$.
$S$ is defined by \autoref{eq:set_of_word_pairs}.

\begin{equation}\label{eq:set_of_word_pairs}
	S = \{(W', W*)|W' = \{w_i\};w_i \in W;W* = W\}
\end{equation}

> Before calculating word probabilities, a sliding window of size $s$ where $s =110$ is used to create a set of sub documents $D_s$ over the document set $D$.

Reviewer 2: ```suggestion
Before calculating word probabilities, a sliding window of size $s$ where $s =110$ is used to create a set of subdocuments $D_s$ over the document set $D$.
```

Author: ```suggestion
Before calculating word probabilities, a sliding window of size $s$ where $s =110$ is used to create a set of sub-documents $D_s$ over the document set $D$.
```

> These sub documents are used rather than the normal documents to capture some degree of words proximity.

Reviewer 2: ```suggestion
These subdocuments are used rather than the normal documents to capture some degree of word proximity.
```

Author: ```suggestion
These sub-documents are used rather than the normal documents to capture some degree of words proximity.
```

> Word probabilities are calculated based on how many documents within the document set $D_s$ that they occur in.

Reviewer 2: ```suggestion
Word probabilities are calculated based on how many documents, within the document set $D_s$, they occur in.
```

> $p(w_i)$, is the number of sub documents in which the word $w_i$ occurs divided by $|D_s|$.

Reviewer 2: ```suggestion
$p(w_i)$ is the number of subdocuments in which the word $w_i$ occurs divided by $|D_s|$.
```

Author: ```suggestion
$p(w_i)$ is the number of sub-documents in which the word $w_i$ occurs divided by $|D_s|$.
```

> $p(w_i, w_j)$, is the number of sub documents in which both words occur divided by $|D_s|$. 

Reviewer 2: ```suggestion
$p(w_i, w_j)$ is the number of subdocuments in which both words occur divided by $|D_s|$. 
```

Author: ```suggestion
$p(w_i, w_j)$ is the number of sub-documents in which both words occur divided by $|D_s|$. 
```

We create a \gls{npmi} matrix of size $|W|\times|W|$, with one entry word pair combination in $W$.
\begin{equation}\label{eq:coherence_2}
	\text{NPMI}(w_i,w_j) =  \frac{\log\frac{p(w_i,w_j) + \epsilon}{p(w_i)\cdot p(w_j)}}{-log(p(w_i,w_j) + \epsilon)}
\end{equation}
\noindent where $\epsilon$ is a low number ($10^{-12}$) used to avoid $log(0)$.

> The \gls{npmi} matrix describes how much each word occurs alone versus together with other words.

Reviewer 2: Må gerne tilføje noget i stil med "Each entry has a value between ? and ?,  where a high value means they occur alone/together with other words."

%This is a -1 to 1 normalization.

After having calculated the \gls{npmi} matrix, we construct context vectors for both elements $W'$ and $W*$ in each word-pair $S_i$, by summing over the rows of the \gls{npmi} matrix.

> This summation describes how much each top word co-occurs with the other words in $W$.

Reviewer 2: Jeg har svært ved, igennem teksten her, at forestille mig hvordan disse vektorer ser ud. Der mangler måske noget for at gøre beskrivelsen tydelig, men jeg ved ikke hvad, da jeg ikke har læst topic coherence artiklen endnu.

Author: jeg ved ikke helt hvad jeg skal skrive tbh :/

Reviewer 2: Det er i orden. Vi kan lade den ligge, og så måske kigge på det senere i skriveprocessen.

\begin{equation}\label{eq:coherence_1}

> 	\overrightarrow{v}(W') = \left\{ \sum_{w_i \in W'} \text{NPMI}(w_i, w_j)^{\gamma} \right\}_{j=1,\dots,|W|}

Reviewer 2: Gamma nævnes ikke i teksten

\end{equation}

We now have a pair of context vectors for each word pair $S_i$ and we want to know how different these vectors are.
This is calculated using cosine similarity as a confirmation measure.
\begin{equation}\label{eq:coherence_3}
	\phi_{S_i}(\overrightarrow{u}, \overrightarrow{w}) = \frac
	{\sum_{i = 1}^{|W|} u_i \cdot w_i}
	{\|\overrightarrow{u}\|_2 \cdot \|\overrightarrow{w}\|_2}
\end{equation}

\noindent where $\overrightarrow{u}$ is the context vector $\overrightarrow{v}(W')$, and $\overrightarrow{w}$ is the context vector $\overrightarrow{v}(W^*)$.

> Lastly, the confirmation measures are aggregated using the arithmetic mean, to achieve the coherence value of topic $t$

Reviewer 2: ```suggestion
Lastly, the confirmation measures are aggregated using the arithmetic mean, to achieve the coherence value of topic $t$.
```

\begin{equation}\label{eq:coherence_4}

> 	C_v = \frac{1}{|S|}\sum_{|S|}^{i=1}\phi_{S_i}

Reviewer 2: ```suggestion
	C_v = \frac{1}{|S|}\sum_{i=1}^{|S|}\phi_{S_i}
```
Kunne ikke finde ligningen i topic coherence artiklen, men det her er vel den korrekte opsætning af summering.

$S$ is defined by \autoref{eq:set_of_word_pairs}.

\begin{equation}\label{eq:set_of_word_pairs}
	S = \{(W', W*)|W' = \{w_i\};w_i \in W;W* = W\}
\end{equation}

Before calculating word probabilities, a sliding window of size $s$ where $s =110$ is used to create a set of subdocuments $D_s$ over the document set $D$.
These subdocuments are used rather than the normal documents to capture some degree of word proximity.
Word probabilities are calculated based on how many documents, within the document set $D_s$, they occur in.
$p(w_i)$ is the number of subdocuments in which the word $w_i$ occurs divided by $|D_s|$.
$p(w_i, w_j)$ is the number of subdocuments in which both words occur divided by $|D_s|$. 

We create a \gls{npmi} matrix of size $|W|\times|W|$, with one entry per word pair combination in $W$.
\begin{equation}\label{eq:coherence_2}
	\text{NPMI}(w_i,w_j) =  \frac{\log\frac{p(w_i,w_j) + \epsilon}{p(w_i)\cdot p(w_j)}}{-log(p(w_i,w_j) + \epsilon)}
\end{equation}
\noindent where $\epsilon$ is a low number ($10^{-12}$) used to avoid $log(0)$.
The \gls{npmi} matrix describes how much each word occurs alone versus together with other words.

> Each value is between $-1$ and $1$, with $-1$ meaning that the words never occour together and 1 meaning that they only occour together.

Reviewer 2: ```suggestion
Each value is between $-1$ and $1$, with $-1$ meaning that the words never occur together and 1 meaning that they only occur together.
```

After having calculated the \gls{npmi} matrix, we construct context vectors for both elements $W'$ and $W*$ in each word-pair $S_i$, by summing over the rows of the \gls{npmi} matrix.
This summation describes how much each top word co-occurs with the other words in $W$.
\begin{equation}\label{eq:coherence_1}
	\overrightarrow{v}(W') = \left\{ \sum_{w_i \in W'} \text{NPMI}(w_i, w_j)^{\gamma} \right\}_{j=1,\dots,|W|}
\end{equation}

> \noindent where gamma can be used to further prioritize higher values. For this paper we use $\gamma = 1$, as recommended by \citet{Syed2017coherence}.

Reviewer 2: ```suggestion
\noindent where $\gamma$ can be used to further prioritize higher values.
For this paper we use $\gamma = 1$, as recommended by \citet{Syed2017coherence}.
```
